### ISSUE TITLE
Add ARM support

### ISSUE THREAD

**Issue by mogopz** [timescale/timescaledb-docker-ha#110]
Hey there,
It'd be really cool if this image was published with ARM support.

Thanks!

**Comment by agustinvinao** [comment]
Im seeing arm version of timescaledb for ARM https://hub.docker.com/r/timescale/timescaledb/tags?page=1&ordering=last_updated but I can't find an ARM version for timescaledb-docker-ha.

I've tried to compile the docker image for ARM but I'm getting an error on `Unable to locate package timescaledb-tools`.

Can you please create a version for ARM64  please. Would be good to have full arm support as timescaledb on dockerhub.

thanks

**Comment by jstallenberger** [comment]
Hey,
It would be great to add ARM images too.
I'm trying to install tobs on a Raspberry to try out the whole stack, but unfortunately I met with the `exec user process caused: exec format error` when initializing the timescaledb pod due to the timescale/timescaledb-ha repository only having amd64 images.
Or is there a workaround for this issue? Are there any k8s Docker images with arm64 compatibility? Can the images from timescale/timescaledb be used on k8s (as they appear to have arm64 support) or not?
Thanks a lot!

**Comment by faberchri** [comment]
A docker image with ARM support and a complete Timescale DB installation (including [TimescaleDB Toolkit](https://github.com/timescale/timescaledb-toolkit)) would help my Organisation a lot in the development process. Let me explain our use case:

Our testing and production databases run on [Timescale cloud](https://portal.timescale.cloud/login). For local development it would sometime be helpful if we could restore DB dumps from prod or test in a Timescale DB running in a local docker container. Pulling a DB dump is easy with [timescaledb-backup](https://github.com/timescale/timescaledb-backup). However when we try to restore the dumps using [the official timescale docker image (timescale/timescaledb)](https://hub.docker.com/r/timescale/timescaledb) we see the following error:
```
2021/09/24 13:50:46 pg_restore: creating EXTENSION "timescaledb_toolkit"
2021/09/24 13:50:46 pg_restore: while PROCESSING TOC:
2021/09/24 13:50:46 pg_restore: from TOC entry 2; 3079 105025 EXTENSION timescaledb_toolkit (no owner)
2021/09/24 13:50:46 pg_restore: error: could not execute query: ERROR:  could not open extension control file "/usr/local/share/postgresql/extension/timescaledb_toolkit.control": No such file or directory
2021/09/24 13:50:46 Command was: CREATE EXTENSION IF NOT EXISTS timescaledb_toolkit WITH SCHEMA public;
2021/09/24 13:50:46 
```
When we use this image ([timescale/timescaledb-ha](https://hub.docker.com/r/timescale/timescaledb-ha)) to start Timescale, restoring the dump works perfectly (because in this image TimescaleDB Toolkit is available, I assume). However, many of us use Apple computers with ARM chips (M1) for development and on these machines this image does not run and in turn many of us can't easily use a test or prod Timescale DB dump for local development.

**Comment by JamesGuthrie** [comment]
I took a look into what it would take to build this image on arm64. The following are my findings:

- postgres does have arm64 images, but _only_ for the current LTS (20.04) see their [announcement](https://www.postgresql.org/about/news/arm64-on-aptpostgresqlorg-2033/)
- The patroni dependency `python3-pysyncobj` is only available on `21.04` and `21.10`
- on `20.04`, `tzdata` wants to be configured interactively, so one must set `DEBIAN_FRONTEND=noninteractive`
- `timescaledb-tools` does not yet have `arm64` images yet, but it looks like support is incoming. I have added a tracking issue in our internal packaging repo https://github.com/timescale/release-build-scripts/issues/35
- currently building the `promscale_extension` fails, but that will be fixed with https://github.com/timescale/timescaledb-docker-ha/pull/181
- currently building `timescaledb-toolkit` will also fail (for the same reason as `promscale_extension`. That will probably be fixed with https://github.com/timescale/timescaledb-docker-ha/pull/182 once it lands.

**Comment by JamesGuthrie** [comment]
@faberchri Can you give some input on this:

> However, many of us use Apple computers with ARM chips (M1) for development and on these machines this image does not run and in turn many of us can't easily use a test or prod Timescale DB dump for local development.

I tried reproducing this issue on my M1 mac, and it seems as though docker now supports cross-arch images:

```
> docker run -e POSTGRES_PASSWORD=password -ti timescale/timescaledb-ha:pg13.5-ts2.5.0-latest
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "C.UTF-8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /home/postgres/pgdata/data ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok
```

Perhaps you need to update to the most recent Docker version? (I'm not sure when they added cross-arch support).

**Comment by faberchri** [comment]
@JamesGuthrie Thanks for investigating this issue.

I can confirm: when running `timescale/timescaledb-ha:pg13.5-ts2.5.1-latest` with `Docker Desktop 4.3.2 (72729)` in emulation mode on Mac M1 the image works as expected.
Importing a timescale database dump with `timescale/timescaledb-backup` worked and I could also query the data from the imported dump using a function from `timescaledb-toolkit` (`average(time_weight(...))`).

**Comment by martinhale** [comment]
Hello! 🙂 

I was just wondering if there were any plans to aim for a native ARM image, as with the [regular Timescale images](https://hub.docker.com/r/timescale/timescaledb/tags).

As suggested above, the latest `-ha` image works well for me on an M1 machine, but it has noticeably slower performance.

Thanks!

**Comment by JamesGuthrie** [comment]
I took a look into this again, I continued work which Feike started in #139. I _think_ that the image can be built (I was able to build it locally on an M1 mac, but some things are different in CI). Unfortunately, GitHub does not have native arm64 runners, so the build is _painfully_ slow, and times out after 6 hours.

One obvious aspect which could be improved is that we're compiling the `toolkit` and `promscale` extensions for each pg version, and it _should_ be possible to install these from a package repository.

**Comment by JamesGuthrie** [comment]
I've opened #240 and #241 to track installing those extensions from packages.

**Comment by boskiv** [comment]
Is that issue related to my issue ? 
I'm trying to run `helm install timescale-release timescale/timescaledb-single -f timescale-values.yaml`

with values
```
image:
  repository: timescale/timescaledb-ha
  tag: pg15.2-ts2.10.3-all
```
in my local minikube cluster running on MBP Apple M1 Max

But in LENS Client I see an error
```
Failed to pull image "timescale/timescaledb-ha:pg15.2-ts2.10.3-all": rpc error: code = Unknown desc = no matching manifest for linux/arm64/v8 in the manifest list entries
```

**Comment by graveland** [comment]
We're getting closer to having a multiplatform image for -ha, and the base image is now a manifest. For now it might be best to use the digest of the exact image you want, ie., [pg15.2-ts2.10.3-all](https://hub.docker.com/r/timescale/timescaledb-ha/tags?page=1&name=pg15.2-ts2.10.3-all) currently has a digest of  `sha256:684ad96abb12bd91112a20db9e79b1afedbc1fd08fbd3ed8407253f7a6f4f108`.

When using docker, adding `--platform amd64` will fetch the correct image (until the arm64 images are pushed that is).

You could also test `timescaledev/timescaledb-ha:pg15.2-ts2.10.3-all`. It's a feature branch and not kept as up-to-date, but feedback would help.

**Comment by aries1980** [comment]
I had a quick play with it a 3-node cluster on a Hetzner  Ampere ARM64, single host. LGTM. 👍

**Comment by aries1980** [comment]
If you need any help, let me know. I'm handy with Github Actions.

**Comment by aries1980** [comment]
Just a heads up, since my last comment, we've been using the logo
`timescaledev/timescaledb-ha:pg14.8-ts2.10.3-all` arm64 build in a smaller development environment. So with great satisfaction. Any ETA to make a stable release of this?

**Comment by leifniem** [comment]
Only using it in local development but it's working so far. Only had to manually enable timescaledb and postgis.

**Comment by RyKilleen** [comment]
Sorry to resurrect an old thread, but I'd personally appreciate a published image with arch support for arm64! I'm happy to help test in whatever capacity.

**Comment by graveland** [comment]
We just got access to a GH beta arm64 runner, so it should be fixed as of #456
